File: csharp/exercises/Webshop.Service/Payment/ThirdPartyPaymentSystemClient.cs
namespace Webshop.Service.Payment;

public class ThirdPartyPaymentSystemClient
{
    public void WaitForConnection()
    {
        // Simulate that we need to connect to a third party system which is not reachable from non-production environment
        Thread.Sleep(TimeSpan.FromDays(365));
    }

    public bool HandlePayment(string userId, decimal amountInNok, string message)
    {
        // Simulate some work that may take some time
        Thread.Sleep(TimeSpan.FromSeconds(5));
        return true;
    }
}

File: java/exercises/src/main/java/Webshop/Service/Payment/ThirdPartyPaymentSystemClient.java
package java.Webshop.Service.Payment;

public class ThirdPartyPaymentSystemClient {
    public void waitForConnection() throws InterruptedException {
        // Simulate that we need to connect to a third-party system which is not reachable from non-production environment
        Thread.sleep(365 * 24 * 60 * 60 * 1000L); // Sleep for 365 days
    }

    public boolean handlePayment(String userId, double amountInNok, String message) throws InterruptedException {
        // Simulate some work that may take some time
        Thread.sleep(5 * 1000); // Sleep for 5 seconds
        return true;
    }
}

